common/context.go

```diff
@@ -23,6 +24,21 @@ func GetHooks(ctx context.Context) *Hooks {
 	return v.(*Hooks)
 }
 
+// WithIterationID adds an identifier for the current iteration to the context.
+func WithIterationID(ctx context.Context, iterID string) context.Context {
+	return context.WithValue(ctx, ctxKeyTraceID, iterID)
+}
+
+// GetIterationID returns the iteration identifier attached to the context.
+func GetIterationID(ctx context.Context) string {
+	v := ctx.Value(ctxKeyTraceID)
+	val, ok := v.(string)
+	if v == nil || !ok {
+		return ""
+	}
+	return val
```

Comment:
The following code will do the same thing:

```suggestion
	s, _ := ctx.Value(ctxKeyIterationID).(string)
	return s // it will return "" on error
```

Reply:
Thanks! We should probably update all the other getters as well then.

Resolved by fd3e409.